internal/coffee/coffee.go

```diff
@@ -86,11 +51,14 @@ func coffeeShopMessage(cs CoffeeSpot) string {
 		lm = fmt.Sprintf("Located on the *%s* side of town.\n", cs.Locale)
 	}
 
-	msg := fmt.Sprintf("Here's a random coffee shop for you to check out! :coffee:\n"+
-		"*%s*\n"+
-		"%s\n"+
-		"Check out the location on Google Maps: %s\n", cs.Name, lm, cs.Location)
-	return msg
+	var sb strings.Builder
```

Comment:
Just curious (asking a Go guru!) - is `strings.Builder` efficient for small batches of string concatenation like this? The .NET counterpart, `StringBuilder`, usually isn't efficient on small scales due to the overhead of pre-allocating a large buffer.

Reply:
Not a guru at all, but I did ask someone at work who is! Essentially everything in `go` is build on what is known as a `slice`, which can be thought of as a dynamic buffer. Therefore, you only utilise the memory you need and it is mutable, so it grows as you use it. Essentially when you write to the Slice, there is a method that checks if there is enough space and if not, grows the slice.